includes/scheduler/class-post.php

```diff
@@ -104,4 +110,41 @@ public static function schedule_post_activity( $new_status, $old_status, $post )
 		// Add the post to the outbox.
 		add_to_outbox( $post, $type, $post->post_author, $content_visibility );
 	}
+
+	/**
+	 * Send announces.
+	 *
+	 * @param int   $outbox_activity_id The outbox activity ID.
+	 * @param array $activity_object    The activity object.
+	 * @param int   $actor_id           The actor ID.
+	 * @param int   $content_visibility The content visibility.
+	 */
+	public static function send_announces( $outbox_activity_id, $activity_object, $actor_id, $content_visibility ) {
+		// Only if we're in both Blog and User modes.
+		if ( ACTIVITYPUB_ACTOR_AND_BLOG_MODE !== \get_option( 'activitypub_actor_mode', ACTIVITYPUB_ACTOR_MODE ) ) {
+			return;
+		}
+
+		// Only if this isn't the Blog Actor.
+		if ( Actors::BLOG_USER_ID === $actor_id ) {
+			return;
+		}
+
+		// Only if the content is public or quiet public.
+		if ( ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC !== $content_visibility ) {
+			return;
+		}
+
+		$activity_type = \get_post_meta( $outbox_activity_id, '_activitypub_activity_type', true );
+
+		// Only if the activity is a Create, Update or Delete.
+		if ( ! in_array( $activity_type, array( 'Create', 'Update', 'Delete' ), true ) ) {
+			return;
+		}
+
+		$transformer = Factory::get_transformer( $activity_object );
+		$activity    = $transformer->to_activity( $activity_type );
+
+		Outbox::add( $activity, 'Announce', Actors::BLOG_USER_ID, ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC );
```

Comment:
Further up we bail if `$actor_id` is the blog user id, here we set it to the blog user id, could you help me understand the difference?

Reply:
If the `$user_id` of the original Activity is the one of the blog user, it should not `Announce` its own Activities.